cime_config/machines/cmake_macros/CNL.cmake

```diff
@@ -1,18 +0,0 @@
-if (NOT MPILIB STREQUAL impi)
-  string(APPEND CMAKE_OPTS " -DCMAKE_SYSTEM_NAME=Catamount")
-endif()
-string(APPEND CPPDEFS " -DLINUX")
-if (COMP_NAME STREQUAL gptl)
-  string(APPEND CPPDEFS " -DHAVE_NANOTIME -DBIT64 -DHAVE_VPRINTF -DHAVE_BACKTRACE -DHAVE_SLASHPROC -DHAVE_COMM_F2C -DHAVE_TIMES -DHAVE_GETTIMEOFDAY")
-endif()
```

Comment:
Note to self: Will check if Frontier/Crusher configurations need these. 
Esp. w.r.t GPTL if they are already present or redundant.

Reply:
@amametjanov We usually have `-DHAVE_SLASHPROC`. What else do we really need?

`-DLINUX` is checked in `shr_sys_mod` conditionals.

Reply:
Grepping through `share`, it appears that only `-DHAVE_VPRINTF -DHAVE_BACKTRACE` can be removed.

Reply:
So I wonder why don't we use rest of the options on other machines too? If we expect almost all machines to have some of these, why don't we move them to defaults in buildlib?

Reply:
Some machines might not have low-level x86 intrinsics like `rdtsc` used by `HAVE_NANOTIME`.
`/proc` was not available on all machines: maybe it is now.
Putting all these configure options in the top-level OS-specific XML/CMake file (instead of a build script) might have been appropriate at the time. Don't know.

Reply:
Might be worth opening an issue on CIME to poll if OS-specific cmake files are useful anywhere.